c_binding/lsm_scsi.c

```diff
@@ -253,30 +205,31 @@ static int _sysfs_vpd83_naa_of_sd_name(char *err_msg, const char *sd_name,
         goto out;
     }
 
-    /*
-     * Check the existence of disk vis /sys/block/sdX folder.
-     */
-    snprintf(sysfs_blk_path, _MAX_SYSFS_BLK_PATH_STR_LEN,
-             _SYSFS_BLK_PATH_FORMAT, sd_name);
-    if (_file_is_exist(err_msg, sysfs_blk_path) == false) {
-        rc = LSM_ERR_NOT_FOUND_DISK;
+    udev = udev_new();
+    if (udev == NULL) {
+        rc = LSM_ERR_NO_MEMORY;
         goto out;
     }
 
-    snprintf(sysfs_path, _MAX_SYSFS_VPD83_PATH_STR_LEN,
-             _SYSFS_VPD83_PATH_FORMAT, sd_name);
-
-    rc = _sysfs_read_file(err_msg, sysfs_path, buff, &read_size,
-                          _MAX_VPD83_PAGE_LEN);
+    sd_udev = udev_device_new_from_subsystem_sysname(udev, "block", sd_name);
+    if (sd_udev == NULL) {
+        rc = LSM_ERR_NOT_FOUND_DISK;
+        goto out;
+    }
 
-    if (rc != LSM_ERR_OK)
+    value = udev_device_get_sysattr_value(sd_udev, "device/vpd_pg83");
```

Comment:
In case you are curious, the reason inconsistent is caused libudev ATA endian problem:

The vpd_pg83 is starting with 00 83(big-edian).
That's why you got SAS disk with empty string while SATA disk with 0x83.